docs/DeveloperGuide.md

```diff
@@ -1,34 +1,600 @@
+
+### 3.4. Model Component
+
+![Model Diagram](images/ModelDiagram.jpg)
+
+### 3.5. Storage Component
+
+The `Storage` class is called by the `HappyPills` class to handle the loading and saving of patient general details, 
+patient medical details and appointment schedule. 
+
+The `loadPatientsFromFile()`, `loadRecordsfromFile()` and `loadAppointmentsFromFile()` methods are used to read saved data from
+local files into the current session of HappyPills. 
+
+The `writePatientsToFile()`, `writeRecordsToFile()` and `writeAppointmentsToFile()` methods write the current state of HappyPills
+into local files by calling them in the `ExitCommand` class.
+
+## 4. Implementation
+
+This section describes some of the details on how the features of the program has been implemented. This section has been broken 
+down into five main features: `Patient Details`, `Patient Medical Records`, `Appointment Scheduling`, `Storage` and `Logging`.
+
+### 4.1. Patient Details Feature
+
+The commands introduced in this feature include : `add`, `edit`, `list`, `delete`, `get`. 
+The commands are implemented with HashMap and use NRIC as key and the Patient class as value.
+The patient list feature is facilitated by PatientMap class which implements the following operations: 
+
+    - PatientMap #add(Patient patient) — This command adds the patient object into the patient list using the patient’s nric as key.
+
+    - PatientMap #remove(String nric) — This command removes the patient object from the existing patient list. 
+
+    - PatientMap #hasKey(String nric) — This command checks whether the patient object resides in the existing patient list. 
+    
+    
+** Design Considerations ** 
+
+*Aspect: Data Structure of the Patient List* 
+
+        Alternative 1 (current choice): Hash Map
+          Pros: Allow faster lookup of patients’ information using the unique identifier (nric)
+          Cons: Implementation is harder and may result in bugs if not implemented accurately.
+            
+        Alternative 2: Array List
+          Pros: This would be easier to implement and retrieve the information.
+          Cons: When a patient is deleted, all the patients in the patient list need to be checked. 
+                This would cause the deletion to be very slow when there is a large number of patients in the list.
+
+#### 4.1.1 Add Patient Details
+
+The user is able to add patients into the program to keep track of the patient's details. 
+The command: 
+
+    add patient /ic S7777777Z /n Alice /p 98765432 /d 01/01/2000 /b B+ /a Peanuts
+    
+will add a patient with `NRIC` as S7777777Z with the following attributes: 
+* name: `Alice`
+* phone number: `98765432`
+* date of birth: `01/01/2000`
+* blood type: `B+`
+* allergies: `Peanuts`
+
+**Implementation** 
+
+The activity diagram below summarises the process of executing an `add` command.
+![Add Patient Sequence Diagram](images/AddPatientSequenceDiagram.jpg)
```

Comment:
![image](https://user-images.githubusercontent.com/51941453/77999354-1df97400-7365-11ea-93f3-3dcf7d4cd3ff.png)
 Does this mean a new parser object and corresponding objects are constructed and later destroyed every time a command is entered?